In [1]:
import pandas as pd

In [5]:
from pathlib import Path

file_path = Path(r"C:/Users/emree/sql-basic-practice/sgk_labor_market_excel_dashboard.xlsm")

file_path.exists()

True

In [7]:
employment_sheet = "25-26-employmentbysector"
df_employment = pd.read_excel(file_path, sheet_name =employment_sheet)

df_employment.head()

,Dönem,Ekonomik Faaliyet,Korunacak/Artacak,Azalacak,Fikrim Yok,Tbl_Anahtar
0,2026Q1,Gayrimenkul Faaliyetleri,0.835,0.029,0.136,2026Q1|Gayrimenkul Faaliyetleri
1,2026Q1,"Mesleki, Bilimsel ve Teknik Faaliyetler",0.824,0.033,0.143,"2026Q1|Mesleki, Bilimsel ve Teknik Faaliyetler"
2,2026Q1,"Kültür, Sanat Eğlence, Dinlence ve Spor",0.819,0.055,0.126,"2026Q1|Kültür, Sanat Eğlence, Dinlence ve Spor"
3,2026Q1,Konaklama ve Yiyecek Hizmeti Faaliyetleri,0.818,0.056,0.126,2026Q1|Konaklama ve Yiyecek Hizmeti Faaliyetleri
4,2026Q1,Eğitim,0.800,0.071,0.129,2026Q1|Eğitim


## Employment by Sector Analysis

This section analyzes employment expectations by economic activity and period.  
The main indicators are the share of responses expecting employment to remain stable/increase, decrease, or expressing no opinion.

In [12]:
df_employment.columns = df_employment.columns.str.strip()

df_employment= df_employment.rename(columns={
    "Dönem" : "period",
    "Ekonomik Faaliyet": "economic_activity",
    "Korunacak/Artacak" : "maintain_or_increase_rate",
    "Azalacak": "decrease_rate",
    "Fikrim Yok" : "no_opinion_rate",
    "Tbl_Anahtar": "Table_Key"
})

df_employment.head()

,period,economic_activity,maintain_or_increase_rate,decrease_rate,no_opinion_rate,Table_Key
0,2026Q1,Gayrimenkul Faaliyetleri,0.835,0.029,0.136,2026Q1|Gayrimenkul Faaliyetleri
1,2026Q1,"Mesleki, Bilimsel ve Teknik Faaliyetler",0.824,0.033,0.143,"2026Q1|Mesleki, Bilimsel ve Teknik Faaliyetler"
2,2026Q1,"Kültür, Sanat Eğlence, Dinlence ve Spor",0.819,0.055,0.126,"2026Q1|Kültür, Sanat Eğlence, Dinlence ve Spor"
3,2026Q1,Konaklama ve Yiyecek Hizmeti Faaliyetleri,0.818,0.056,0.126,2026Q1|Konaklama ve Yiyecek Hizmeti Faaliyetleri
4,2026Q1,Eğitim,0.800,0.071,0.129,2026Q1|Eğitim


### Data Quality Check

Before analysis, missing values and duplicate rows are checked to understand the reliability of the dataset.

In [13]:
df_employment.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 85 entries, 0 to 84
Data columns (total 6 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   period                     85 non-null     object 
 1   economic_activity          85 non-null     object 
 2   maintain_or_increase_rate  85 non-null     float64
 3   decrease_rate              85 non-null     float64
 4   no_opinion_rate            85 non-null     float64
 5   Table_Key                  85 non-null     object 
dtypes: float64(3), object(3)
memory usage: 4.1+ KB


In [14]:
df_employment.isnull().sum()

period                       0
economic_activity            0
maintain_or_increase_rate    0
decrease_rate                0
no_opinion_rate              0
Table_Key                    0
dtype: int64

In [16]:
df_employment.duplicated().sum()

np.int64(0)

### Analysis 1: Period-Level Employment Expectation Summary

This analysis summarizes average employment expectations by period.  
It helps compare whether employment expectations are becoming more positive or negative over time.

In [18]:
employment_period_summary = df_employment.groupby("period").agg(
    average_maintain_or_increase_rate = ("maintain_or_increase_rate", "mean"),
    average_decrease_rate = ("decrease_rate", "mean"),
    average_no_opinion_rate = ("no_opinion_rate", "mean")
).reset_index()

employment_period_summary

,period,average_maintain_or_increase_rate,average_decrease_rate,average_no_opinion_rate
0,2025Q1,0.769471,0.055824,0.168824
1,2025Q2,0.774118,0.042882,0.183235
2,2025Q3,0.774588,0.049118,0.176118
3,2025Q4,0.783176,0.047176,0.169765
4,2026Q1,0.784765,0.055294,0.159647


### Analysis 2: Sectors with the Highest Maintain or Increase Expectation

This analysis identifies the sectors with the highest expectation that employment will be maintained or increased.

In [19]:
top_maintain_or_increase_rate = df_employment [
    ["period", "economic_activity", "maintain_or_increase_rate", "decrease_rate", "no_opinion_rate"]
].sort_values(by="maintain_or_increase_rate", ascending = False)

top_maintain_or_increase_rate.head(10)

,period,economic_activity,maintain_or_increase_rate,decrease_rate,no_opinion_rate
51,2025Q2,İnsan Sağlığı ve Sosyal Hizmet Faaliyetleri,0.861,0.034,0.105
34,2025Q3,İnsan Sağlığı ve Sosyal Hizmet Faaliyetleri,0.855,0.024,0.121
35,2025Q3,Finans ve Sigorta Faaliyetleri,0.846,0.034,0.120
17,2025Q4,"Mesleki, Bilimsel ve Teknik Faaliyetler",0.844,0.030,0.126
52,2025Q2,Diğer Hizmet Faaliyetleri,0.840,0.026,0.134
18,2025Q4,Finans ve Sigorta Faaliyetleri,0.840,0.041,0.119
36,2025Q3,"Mesleki, Bilimsel ve Teknik Faaliyetler",0.839,0.026,0.135
37,2025Q3,Diğer Hizmet Faaliyetleri,0.836,0.027,0.137
0,2026Q1,Gayrimenkul Faaliyetleri,0.835,0.029,0.136
19,2025Q4,İnsan Sağlığı ve Sosyal Hizmet Faaliyetleri,0.835,0.034,0.132


### Analysis 3: Sectors with the Highest Decrease Expectation

This analysis identifies the sectors where employment decrease expectation is the highest.

In [20]:
top_decrease_rate = df_employment[
    ["period", "economic_activity", "maintain_or_increase_rate", "decrease_rate", "no_opinion_rate"]
].sort_values(by="decrease_rate", ascending = False)

top_decrease_rate.head(10)

,period,economic_activity,maintain_or_increase_rate,decrease_rate,no_opinion_rate
62,2025Q2,Eğitim,0.738,0.107,0.155
84,2025Q1,"Su Temini; Kanalizasyon, Atık Yönetimi ve İyil...",0.671,0.105,0.211
33,2025Q4,Madencilik ve Taş Ocakçılığı,0.646,0.081,0.273
32,2025Q4,"Su Temini; Kanalizasyon, Atık Yönetimi ve İyil...",0.718,0.081,0.201
82,2025Q1,İmalat,0.726,0.079,0.218
49,2025Q3,"Su Temini; Kanalizasyon, Atık Yönetimi ve İyil...",0.686,0.076,0.237
13,2026Q1,İmalat,0.769,0.076,0.154
16,2026Q1,"Su Temini; Kanalizasyon, Atık Yönetimi ve İyil...",0.731,0.071,0.198
4,2026Q1,Eğitim,0.800,0.071,0.129
48,2025Q3,İmalat,0.705,0.071,0.225


### Analysis 4: Sectors with the Highest Uncertainty

This analysis identifies the sectors with the highest no-opinion rate.  
A higher no-opinion rate may indicate uncertainty about future employment expectations.

In [23]:
top_uncertainty = df_employment[
    ["period","economic_activity","maintain_or_increase_rate","decrease_rate","no_opinion_rate"]
].sort_values(by="no_opinion_rate", ascending = False)

top_uncertainty.head(10)

,period,economic_activity,maintain_or_increase_rate,decrease_rate,no_opinion_rate
33,2025Q4,Madencilik ve Taş Ocakçılığı,0.646,0.081,0.273
67,2025Q2,İnşaat,0.695,0.043,0.263
50,2025Q3,Madencilik ve Taş Ocakçılığı,0.677,0.066,0.257
63,2025Q2,"Elektrik, Gaz, Buhar ve İklimlendirme Üretimi ...",0.730,0.025,0.245
49,2025Q3,"Su Temini; Kanalizasyon, Atık Yönetimi ve İyil...",0.686,0.076,0.237
66,2025Q2,Madencilik ve Taş Ocakçılığı,0.708,0.056,0.236
48,2025Q3,İmalat,0.705,0.071,0.225
47,2025Q3,İnşaat,0.708,0.070,0.221
82,2025Q1,İmalat,0.726,0.079,0.218
65,2025Q2,"Su Temini; Kanalizasyon, Atık Yönetimi ve İyil...",0.717,0.066,0.217


### Analysis 5: Employment Outlook Score

This analysis creates a simple employment outlook score by subtracting the decrease expectation rate from the maintain/increase expectation rate.  
A higher score indicates a more positive employment expectation.

In [26]:
df_employment["employment_outlook_score"] = (
    df_employment["maintain_or_increase_rate"] - df_employment["decrease_rate"]
)

employment_outlook = df_employment[
    ["period", "economic_activity", "maintain_or_increase_rate", "decrease_rate", "no_opinion_rate", "employment_outlook_score"]
].sort_values(by="employment_outlook_score", ascending = False)

employment_outlook.head(10)

,period,economic_activity,maintain_or_increase_rate,decrease_rate,no_opinion_rate,employment_outlook_score
34,2025Q3,İnsan Sağlığı ve Sosyal Hizmet Faaliyetleri,0.855,0.024,0.121,0.831
51,2025Q2,İnsan Sağlığı ve Sosyal Hizmet Faaliyetleri,0.861,0.034,0.105,0.827
17,2025Q4,"Mesleki, Bilimsel ve Teknik Faaliyetler",0.844,0.030,0.126,0.814
52,2025Q2,Diğer Hizmet Faaliyetleri,0.840,0.026,0.134,0.814
69,2025Q1,"Elektrik, Gaz, Buhar ve İklimlendirme Üretimi ...",0.830,0.017,0.154,0.813
36,2025Q3,"Mesleki, Bilimsel ve Teknik Faaliyetler",0.839,0.026,0.135,0.813
35,2025Q3,Finans ve Sigorta Faaliyetleri,0.846,0.034,0.120,0.812
37,2025Q3,Diğer Hizmet Faaliyetleri,0.836,0.027,0.137,0.809
0,2026Q1,Gayrimenkul Faaliyetleri,0.835,0.029,0.136,0.806
53,2025Q2,"Mesleki, Bilimsel ve Teknik Faaliyetler",0.827,0.022,0.151,0.805


## Key Insights

- Employment expectations are generally positive across periods, with average maintain/increase rates around 77%–78%.
- The average maintain/increase expectation reaches its highest level in 2026 Q1.
- Human Health and Social Work Activities shows one of the strongest positive employment outlooks.
- Education and Water Supply, Sewerage and Waste Management appear among the sectors with relatively higher decrease expectations in certain periods.
- Mining and Quarrying and Construction show higher no-opinion rates in some periods, indicating stronger uncertainty.
- The employment outlook score provides a clearer comparison by combining positive and negative employment expectations into a single metric.